src/core/profile_manager.mjs

```diff
@@ -0,0 +1,317 @@
+    let accountId = constants.HEDERA_NODE_ACCOUNT_ID_START.num
+
+    // set consensus pod level resources
+    for (let nodeIndex = 0; nodeIndex < nodeIds.length; nodeIndex++) {
+      this._setValue(`hedera.nodes.${nodeIndex}.name`, nodeIds[nodeIndex], yamlRoot)
+      this._setValue(`hedera.nodes.${nodeIndex}.accountId`, `${realm}.${shard}.${accountId++}`, yamlRoot)
+      this._setChartResources(`hedera.nodes.${nodeIndex}`, profile.consensus, yamlRoot)
+    }
+
+    if (profile.consensus) {
+      // set default for consensus pod
+      this._setChartResources('defaults.root', profile.consensus.root, yamlRoot)
+
+      // set sidecar resources
+      for (const sidecar of consensusSidecars) {
+        this._setChartResources(`defaults.sidecars.${sidecar}`, profile.consensus[sidecar], yamlRoot)
+      }
+    }
+
+    return yamlRoot
+  }
+
+  resourcesForHAProxyPod (profile, yamlRoot) {
+    if (!profile) throw new MissingArgumentError('profile is required')
+    if (!profile.haproxy) return // use chart defaults
+
+    return this._setChartResources('defaults.haproxy', profile.haproxy, yamlRoot)
+  }
+
+  resourcesForEnvoyProxyPod (profile, yamlRoot) {
+    if (!profile) throw new MissingArgumentError('profile is required')
+    if (!profile.envoyProxy) return // use chart defaults
+    return this._setChartResources('defaults.envoyProxy', profile.envoyProxy, yamlRoot)
+  }
+
+  resourcesForHederaExplorerPod (profile, yamlRoot) {
+    if (!profile) throw new MissingArgumentError('profile is required')
+    if (!profile.explorer) return
+    return this._setChartResources('hedera-explorer', profile.explorer, yamlRoot)
+  }
+
+  resourcesForMinioTenantPod (profile, yamlRoot) {
+    if (!profile) throw new MissingArgumentError('profile is required')
+    if (!profile.minio || !profile.minio.tenant) return // use chart defaults
+
+    for (const poolIndex in profile.minio.tenant.pools) {
+      const pool = profile.minio.tenant.pools[poolIndex]
+      for (const prop in pool) {
+        if (prop !== 'resources') {
+          this._setValue(`minio-server.tenant.pools.${poolIndex}.${prop}`, pool[prop], yamlRoot)
+        }
+      }
+
+      this._setChartResources(`minio-server.tenant.pools.${poolIndex}`, pool, yamlRoot)
+    }
+
+    return yamlRoot
+  }
+
+  /**
+   * Prepare a values file for FST Helm chart
+   * @param profileName resource profile name
+   * @return {Promise<string>} return the full path to the values file
+   */
+  prepareValuesForFSTChart (profileName) {
+    if (!profileName) throw new MissingArgumentError('profileName is required')
+    const profile = this.getProfile(profileName)
+
+    const nodeIds = helpers.parseNodeIDs(this.configManager.getFlag(flags.nodeIDs))
+    if (!nodeIds) throw new FullstackTestingError('Node IDs are not set in the config')
+
+    // generate the yaml
+    const yamlRoot = {}
+    this.resourcesForConsensusPod(profile, nodeIds, yamlRoot)
+    this.resourcesForHAProxyPod(profile, yamlRoot)
```

Comment:
```suggestion
    this.resourcesForHaProxyPod(profile, yamlRoot)
```

Reply:
ok, Thanks, I'll make the chages with the rest of the updates.

Reply:
Refactored the method and updated other occurrances.